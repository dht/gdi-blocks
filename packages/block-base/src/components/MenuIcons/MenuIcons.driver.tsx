import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MenuIcons, MenuIconsProps } from './MenuIcons';
import { BaseComponentDriver } from 'testing-base';

export class MenuIconsDriver extends BaseComponentDriver {
    private props: Partial<MenuIconsProps> = {};

    constructor() {
        super('MenuIcons');
    }

    when: any = {
        rendered: () => {
            render(<MenuIcons {...(this.props as MenuIconsProps)} />);
            return this;
        },
        clicked: () => {
            fireEvent.click(this.wrapper);
            return this;
        },
        snapshot: () => {
            return this.snapshot(<MenuIcons {...(this.props as MenuIconsProps)} />);
        },
    };

    given: any = {
        props: (props: Partial<MenuIconsProps>) => {
            this.props = props;
            return this;
        },
    };

    get = {
        WrapperClassName: () => {
            return this.wrapper.className;
        },
        label: () => {
            return this.wrapper.innerHTML;
        },
    };
}
