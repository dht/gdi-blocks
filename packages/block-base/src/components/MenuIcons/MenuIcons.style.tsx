import styled from 'styled-components';

export const Wrapper = styled.div`
    display: flex;
    flex-direction: row;
    align-items: center;
`;

export const Item = styled.div`
    margin: 0 2px;
    padding: 0 5px;

    i {
        color: #334;
        font-size: 30px;
    }
`;
