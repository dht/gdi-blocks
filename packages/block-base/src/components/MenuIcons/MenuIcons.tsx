import React from 'react';
import { Item, Wrapper } from './MenuIcons.style';

export type MenuIconsProps = {};

export function MenuIcons(_props: MenuIconsProps) {
    return (
        <Wrapper className='MenuIcons-wrapper' data-testid='MenuIcons-wrapper'>
            <Item>
                <i className='material-icons'>image</i>
            </Item>
            <Item>
                <i className='material-icons'>tag</i>
            </Item>
        </Wrapper>
    );
}

export default MenuIcons;
