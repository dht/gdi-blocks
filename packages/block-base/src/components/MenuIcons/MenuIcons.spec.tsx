import { MenuIconsDriver } from './MenuIcons.driver';
import Chance from 'chance';

const chance = new Chance();

describe('MenuIcons', () => {
    let driver: MenuIconsDriver;

    beforeAll(() => {
        driver = new MenuIconsDriver();
    });

    it('should render button', () => {
        const label = chance.word();

        const element = driver.given
            .props({
                title: label,
            })
            .when.rendered();

        const wrapperClassName = element.get.wrapperClassName();
        const innerText = element.get.label();

        expect(wrapperClassName).toContain('MenuIcons-wrapper');
        expect(innerText).toBe(label);
    });

    it('should click button', () => {
        const callback = jest.fn();

        driver.given
            .props({
                onClick: callback,
            })
            .when.rendered()
            .when.clicked();

        expect(callback).toHaveBeenCalledTimes(1);
    });
});

describe('MenuIcons snapshots', () => {
    let driver: MenuIconsDriver;

    beforeAll(() => {
        driver = new MenuIconsDriver();
    });

    it('should match snapshot', () => {
        expect(driver.when.snapshot()).toMatchSnapshot();
    });
});
