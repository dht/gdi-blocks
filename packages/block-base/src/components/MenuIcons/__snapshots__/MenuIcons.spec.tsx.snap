// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuIcons snapshots should match snapshot 1`] = `
<div
  className="sc-bcXHqe eNtJIX MenuIcons-wrapper"
  data-testid="MenuIcons-wrapper"
>
  <div
    className="sc-gswNZR fHsAle"
  >
    <i
      className="material-icons"
    >
      image
    </i>
  </div>
  <div
    className="sc-gswNZR fHsAle"
  >
    <i
      className="material-icons"
    >
      tag
    </i>
  </div>
</div>
`;
